azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -0,0 +1,494 @@
+
+import os
+from typing import List, Optional, Tuple
+
+from azure.core.paging import PageIterator
+from azure.core.exceptions import ResourceNotFoundError
+from knack.log import get_logger
+from rich.console import Console
+import yaml
+
+from azext_edge.edge.providers.orchestration.common import CUSTOM_LOCATIONS_API_VERSION
+from azext_edge.edge.providers.orchestration.resources.instances import Instances
+from azext_edge.edge.util.file_operations import read_file_content, validate_file_extension
+from azext_edge.edge.util.queryable import Queryable
+from azext_edge.edge.util.az_client import (
+    parse_resource_id,
+    get_keyvault_client,
+    get_ssc_mgmt_client,
+    wait_for_terminal_state,
+)
+
+logger = get_logger(__name__)
+
+console = Console()
+
+OPCUA_SPC_NAME = "opc-ua-connector"
+OPCUA_TRUST_LIST_SECRET_SYNC_NAME = "aio-opc-ua-broker-trust-list"
+OPCUA_ISSUER_LIST_SECRET_SYNC_NAME = "aio-opc-ua-broker-issuer-list"
+OPCUA_CLIENT_CERT_SECRET_SYNC_NAME = "aio-opc-ua-broker-client-certificate"
+SERVICE_ACCOUNT_NAME = "aio-ssc-sa"
+
+
+class OpcUACerts(Queryable):
+
+    def __init__(self, cmd):
+        super().__init__(cmd=cmd)
+        self.instances = Instances(self.cmd)
+        self.ssc_mgmt_client = get_ssc_mgmt_client(
+            subscription_id=self.default_subscription_id,
+        )
+        self
+
+    def trust_add(self, instance_name: str, resource_group: str, file: str, secret_name: Optional[str] = None) -> dict:
+        cl_resources = self._get_cl_resources(instance_name=instance_name, resource_group=resource_group)
+        secretsync_spc = self._find_existing_spc(instance_name=instance_name, cl_resources=cl_resources)
+        if not secretsync_spc:
+            return
+
+        # get properties from default spc
+        spc_properties = secretsync_spc.get("properties", {})
+        spc_keyvault_name = spc_properties.get("keyvaultName", "")
+        spc_tenant_id = spc_properties.get("tenantId", "")
+        spc_client_id = spc_properties.get("clientId", "")
+
+        self.keyvault_client = get_keyvault_client(
+            subscription_id=self.default_subscription_id,
+            keyvault_name=spc_keyvault_name,
+        )
+
+        secrets: PageIterator = self.keyvault_client.list_properties_of_secrets()
+
+        # get file extension
+        file_name = os.path.basename(file)
+        # get cert name by removing extension and path in front
+        cert_extension = validate_file_extension(file_name, [".der", ".crt"])
+
+        secret_name = secret_name if secret_name else file_name.replace(".", "-")
+
+        # iterate over secrets to check if secret with same name exists
+        secret_name = self._check_and_update_secret_name(secrets, secret_name, spc_keyvault_name)
+        self._upload_to_key_vault(secret_name, file, cert_extension)
+
+        # check if there is a spc called "opc-ua-connector", if not create one
```

Comment:
does it need to have that exact name? If so that should be documented somewhere for the user (could be in the ms docs)

Reply:
if you are creating things - make sure to note them in the long description